game/src/main/scala/io/aigar/game/Vector2Utils.scala

```diff
@@ -0,0 +1,11 @@
+package io.aigar.game
+
+import com.github.jpbetz.subspace._
+
+object Vector2Utils {
+  implicit class StateAddon(val vector: Vector2) {
```

Comment:
As in why it's there, or what that does in Scala?

We're [adding a method to `Vector2`](http://alvinalexander.com/scala/scala-how-to-add-new-methods-to-existing-classes), since we're using an existing class (`Vector2`) instead of creating a custom `Position` class. The result is that we can just use a similar `.state` interface in every implementation of `.state` (they all look very similar thanks to this). E.g. the implementation of `Cell.state` stays similar to the other implementations of `.state` instead of doing something custom for `Vector2`.